server/token/token.go

```diff
@@ -0,0 +1,76 @@
+package token
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/lesterfernandez/roommate-finder/server/data"
+	"net/http"
+	"time"
+)
+
+var JWTKey = []byte("secret")
+
+//=====================================================================================
+// Encode JWT into JSON, send it as response
+//=====================================================================================
+
+func SendJWT(w http.ResponseWriter, jwt string) {
+	w.Header().Set("Application-Type", "application/json")
+	json.NewEncoder(w).Encode(data.TokenMessage{Token: jwt})
+}
+
+//=====================================================================================
+// Create JWT
+//=====================================================================================
+
+func CreateJWT(username string) (string, error) {
+
+	expirationTime := time.Now().Add(time.Minute * 1)
+
+	claims := &jwt.RegisteredClaims{
+		Subject:   username,
+		ExpiresAt: jwt.NewNumericDate(expirationTime),
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	tokenString, err := token.SignedString(JWTKey)
+
+	return tokenString, err
+}
+
+//========================================================================================================
+// Verify that a JWT is valid, and return the registeredClaims
+//========================================================================================================
+
+func VerifyJWT(tokenString string, secretKey []byte) (jwt.MapClaims, error) {
+
+	//Verify that the signing method is correct
+	verifySigningMethod := func(token *jwt.Token) (interface{}, error) {
+		// Check the signing method
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
```

Comment:
I notice you are checking the token's signing method through a type assertion and checking if it succeeded. This should work perfectly fine! However, the `jwt.Parse` function takes in `ParserOption` as its last arguments. You could pass in a `WithValidMethods` parser option to automatically check the signing method. It would make the code more readable.

Here's the link: https://pkg.go.dev/github.com/golang-jwt/jwt/v5#Parse
In the function definition you can click on `ParserOption` in the function definition to learn more about that.

After this change, you can remove the check from the verification function.